src/documents/models.py

```diff
@@ -650,6 +650,11 @@ class PaperlessTask(ModelWithOwner):
     ALL_STATES = sorted(states.ALL_STATES)
     TASK_STATE_CHOICES = sorted(zip(ALL_STATES, ALL_STATES))
 
+    class TaskType(models.TextChoices):
```

Comment:
Alongside something like this, why don't we update `task_name` to accept certain choices as we add support for some other types besides `tasks.consume_file`?

Then perhaps a trigger type field for scheduled, manual, auto?

Reply:
Ive changed this part in the meantime, lmk about the rest (whenever)